### PR TITLE
Implement scalable import and correct export functionality

### PR DESCRIPTION
**Pull request by flyingsilverfin**
## What is the goal of this PR?

We fix some stability issues in the import and export process that were indicated by #6310. Imports are now also multi-threaded, multi-transactional, and can scales to huge imports without OOM.

## What are the changes implemented in this PR?
* correctly clean up session/transaction resources at the end of an import
* rewrite `Import` to be multi-threaded and write ID mappings to a rocksDB instance to avoid running out of memory
* since on commit new IIDs are rewritten, we expose the mapping from buffered to persisted IIDs as a `RocksTransaction` API
* add missing header to exported data file
* use `toHexString()` for IID encoding, instead of `encodeString()` -- this is because UTF_8 encoding converts invalid bytes sequences (which we have in our IID byte arrays) with standard sequences. This can cause corruption because two concepts can end up with the same serialised string ID
*  make progress logging more granular with individual types tracked
* replace the `//test/integration/migration/data.typedb` file with a correct export containing both header and non-clashing IDs
* update and remove most rules from the test schema to speed up the test execution (rule validation can take around 20 seconds)

### CODE REVIEW

**Comment by flyingsilverfin** [comment]
TODO: we can still end up an error state if more than one BATCH_SIZE are imported in the transaction, because of the ID rewrites we do

File: migrator/DataImporter.java [hunk 1]
```diff
@@ -34,286 +35,539 @@
...
-                    attribute = attributeType.asLong().put(valueMsg.getLong());
-                    break;
-                case DOUBLE:
-                    attribute = attributeType.asDouble().put(valueMsg.getDouble());
-                    break;
-                case DATETIME:
-                    attribute = attributeType.asDateTime().put(
-                            Instant.ofEpochMilli(valueMsg.getDatetime()).atZone(ZoneId.of("Z")).toLocalDateTime());
-                    break;
+    private class InitAndAttributes extends ImportWorker {
```
Comment (flyingsilverfin) [inline migrator/DataImporter.java]: in the first pass of the file we load attributes, and search for the checksum at the end of the file

File: test/integration/migrator/MigratorTest.java [hunk 2]
```diff
@@ -48,7 +48,7 @@
     private static final Path logDir = dataDir.resolve("logs");
     private static final Database options = new Database().dataDir(dataDir).logsDir(logDir);
     private static final String database = "typedb";
-    private static final Path schemaPath = Paths.get("test/integration/migrator/schema.gql");
+    private static final Path schemaPath = Paths.get("test/integration/migrator/schema.tql");
     private final Path dataPath = Paths.get("test/integration/migrator/data.typedb");
```
Comment (lolski) [inline test/integration/migrator/MigratorTest.java:52]: we're adding this file `test/integration/migrator/data.typedb 
`, of size 6.25MB, intentionally, right?

Comment (flyingsilverfin) [reply test/integration/migrator/MigratorTest.java:52]: yes, wanted something a bit more complete

File: migrator/BUILD [hunk 3]
```diff
@@ -63,7 +70,17 @@ native_java_libraries(
     native_libraries_deps = [
         "//:typedb",
         "//concept:concept",
-        "//logic:logic"
+        "//logic:logic",
+        "//rocks:rocks",
```
Comment (flyingsilverfin) [inline migrator/BUILD:74]: awkward to depend on rocks - should we be putting the APIs on the Transaction interface that we require?

**Pull request merged**